java/vector/src/main/java/org/apache/arrow/vector/testing/ValueVectorDataPopulator.java

```diff
@@ -0,0 +1,542 @@
+/*
```

Comment:
should this go under src/test?

Reply:
thanks, previous version was under src/test, to make it reusable for other module tests, @emkornfield suggested moving it to main, and we are waiting for @jacques-n to confirm. see https://github.com/apache/arrow/pull/5693#discussion_r338893344

Reply:
Ok, I'll let @jacques-n make the call, but my preference would be to not include this in the main artifact. There should be some other way for additional modules to use this in tests, if needed.

Reply:
I think the steps described in https://cleantestcode.wordpress.com/2014/04/13/reusable-test-utility-code-and-maven/ make sense to keep it out of main.  (I prefer the naming "testing" to testutils)

Reply:
I followed the steps in the link you provided, but failed to work...

Reply:
can you provide more context on didn't work?

Reply:
mvn clean install failed which says could not found this class. I didn't keep the original message.

Reply:
Hmm, OK, based on https://maven.apache.org/plugins/maven-jar-plugin/examples/create-test-jar.html it seems like we really should create a new module for this code (or you can try to follow the first example, but I think that is the same as the one I linked above).

Reply:
Hmm, I tried to follow the first example and it worked :), now the utility class is under test package and I add a test case in Avro module to make sure it is reusable.
Please see if it looks reasonable @emkornfield